lib/response.js

```diff
@@ -1052,6 +1052,32 @@ function sendfile(res, file, options, callback) {
 }
 
 /**
+ * Run every tracers set at the application level for given events
+ * and arguments. To set a tracer on the application use the `instrument`
+ * method.
+ *
+ * @param {function} event The event to trace.
+ * @return {ServerResponse} for chaining
+ * @public
+ */
+
+res.trace = function(event) {
+  var args = [];
+  switch (arguments.length) {
+    case 0:
+      throw new Error('No event defined!');
+    case 1: break;
+    case 2:
+      args = [arguments[1]];
+      break;
+    default:
+      args = [].slice.call(arguments, 1);
+  }
+  this.app._callTracers(this, event, args);
```

Comment:
@dougwilson Ah yes I see what you mean - and I agree something like `process.nextTick(tracers[i], options);` @ https://github.com/expressjs/express/pull/2437/files/786785b77061bffff90c7b710475af24a35e7ff5#diff-5372f626ee15242f1e2c6eb31655b4faR678 
As the event info has already been recorded your point makes sense.  Can you see any problems @frankrousseau ? 

In terms of use cases the main one (Recording request times) is the main driver and the detail is discussed here 
https://github.com/expressjs/express/issues/2191#issuecomment-62252917 
Providing an abstraction where people can bring there own providers which was the main objection raised in that thread and I think this requests covers it off. 
Another upside of this approach it that developing with and using third party middleware will improve as people will be able to depend on res.trace being there which will improve debugging/trouble shooting. 
My experience with 3rd Party middleware is that people either provide no trace info or bring whatever flavor of logger they see fit and that can be troublesome in terms of integration.

In terms of docs I would be more than happy to flesh out those use cases with @crandmck and support anyway I can.